plugins/forwarder/kafka/nativetracing/forwarder.go

```diff
@@ -0,0 +1,121 @@
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package nativetracing
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/Shopify/sarama"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/apache/skywalking-satellite/internal/pkg/config"
+	"github.com/apache/skywalking-satellite/internal/pkg/log"
+	"github.com/apache/skywalking-satellite/internal/satellite/event"
+
+	v3 "skywalking.apache.org/repo/goapi/collect/language/agent/v3"
+	v1 "skywalking.apache.org/repo/goapi/satellite/data/v1"
+)
+
+const (
+	Name     = "native-tracing-kafka-forwarder"
+	ShowName = "Native Tracing Kafka Forwarder"
+)
+
+type Forwarder struct {
+	config.CommonFields
+	Topic    string `mapstructure:"topic"` // The forwarder topic.
+	producer sarama.SyncProducer
+}
+
+func (f *Forwarder) Name() string {
+	return Name
+}
+
+func (f *Forwarder) ShowName() string {
+	return ShowName
+}
+
+func (f *Forwarder) Description() string {
+	return "This is a synchronization Kafka forwarder with the SkyWalking native log protocol."
+}
+
+func (f *Forwarder) DefaultConfig() string {
+	return `
+# The remote topic. 
+topic: "skywalking-segments"
+`
+}
+
+func (f *Forwarder) Prepare(connection interface{}) error {
+	client, ok := connection.(sarama.Client)
+	if !ok {
+		return fmt.Errorf("the %s only accepts a grpc client, but received a %s",
+			f.Name(), reflect.TypeOf(connection).String())
+	}
+	producer, err := sarama.NewSyncProducerFromClient(client)
+	if err != nil {
+		return err
+	}
+	f.producer = producer
+	return nil
+}
+
+func (f *Forwarder) Forward(batch event.BatchEvents) error {
+	var message []*sarama.ProducerMessage
+	for _, e := range batch {
+		switch data := e.GetData().(type) {
+		case *v1.SniffData_Segment:
+			segmentObject := &v3.SegmentObject{}
+			err := proto.Unmarshal(data.Segment, segmentObject)
+			if err != nil {
+				return err
+			}
+
+			message = append(message, &sarama.ProducerMessage{
+				Topic: f.Topic,
+				Key:   sarama.StringEncoder(segmentObject.GetTraceSegmentId()),
+				Value: sarama.ByteEncoder(data.Segment),
+			})
+		case *v1.SniffData_SpanAttachedEvent:
+			// SniffData_SpanAttachedEvent is from ebpf agent, skywalking-rover project.
+			// You could find it here,
+			// https://github.com/apache/skywalking-data-collect-protocol/blob/0da9c8b3e111fb51c9f8854cae16d4519462ecfe
+			// /language-agent/Tracing.proto#L244
+			// ref: https://github.com/apache/skywalking-satellite/pull/128#discussion_r1136909393
+			log.Logger.WithField("pipe", f.PipeName).Warnf("native-tracing-kafka-forwarder " +
```

Comment:
Why there still not implemented?

Reply:
> Can you confirm whether e2e can be run on WSL?

Using Linux and macOS would be much much better than WSL on Windows. That is pain.

Reply:
Do you really not support WSL? If that's the case, then I'll have to switch to that Mac. PS: I'm not very used to using Mac! :(

Reply:
It should be works well with windows. Could you provide more detail about the error?

Reply:
We don't test on WSL. No one in the maintainer team is using Windows. That is the point.  Linux is very easy to set up. You could run from there on a Windows.

Reply:
> It should be works well with windows. Could you provide more detail about the error?

the example of skywalking-infra-e2e:

1，windows CMD:
```
F:\github\apache\skywalking-infra-e2e\bin\windows>.\e2e.exe  run -c ../../examples/compose/e2e.yaml
INFO load the e2e config successfully             
: invalid reference format: no such image: apache/skywalking-oap-server:8.5.0-es6
INFO deleting docker compose cluster...           
INFO cleanup part finished successfully           
ERROR Local Docker compose exited abnormally whilst running docker-compose.exe: [--env-file F:\github\apache\skywalking-infra-e2e\examples\compose\env up -d]. exit status 1 
```

2，windows PowerShell:
```
PS F:\github\apache\skywalking-infra-e2e\bin\windows> .\e2e.exe  run -c ../../examples/compose/e2e.yaml
INFO load the e2e config successfully
: invalid reference format: no such image: apache/skywalking-ui:8.5.0
INFO deleting docker compose cluster...
INFO cleanup part finished successfully
ERROR Local Docker compose exited abnormally whilst running docker-compose.exe: [--env-file F:\github\apache\skywalking-infra-e2e\examples\compose\env up -d]. exit status 1
```

3，WSL 2:
```
root@DESKTOP-P1D7M3U:/mnt/f/github/apache/skywalking-infra-e2e/bin/linux# e2e run -c ../../examples/compose/e2e.yaml
INFO load the e2e config successfully
: invalid reference format: no such image: apache/skywalking-ui:8.5.0
INFO deleting docker compose cluster...
INFO cleanup part finished successfully
ERROR Local Docker compose exited abnormally whilst running docker-compose: [--env-file /mnt/f/github/apache/skywalking-infra-e2e/examples/compose/env up -d]. exit status 1
```



docker-compose :
```
root@DESKTOP-P1D7M3U:/mnt/f/github/apache/skywalking-infra-e2e/examples/compose# docker-compose --env-file=./env  up -d
[+] Running 2/2
 ⠿ Container compose-oap-1  Healthy                                                                                                                                                                                                                                  11.6s 
 ⠿ Container compose-ui-1   Started  
```

I have been able to successfully start using the docker-compose command on both Windows and WSL. However, I have never been successful in using the e2e command directly.

Reply:
> We don't test on WSL. No one in the maintainer team is using Windows. That is the point. Linux is very easy to set up. You could run from there on a Windows.

I'm going to try installing a Linux desktop on Hyper-V.

Reply:
> It should be works well with windows.

The issue was caused by LF and CRLF. To fix this,   add .sh, env, and mvnw to the .gitattributes file in Git. This will prevent them from being converted to CRLF (\r\n) when cloned.   then use dos2unix or Goland (file->file properties->Line separators->LF unix and macOS(\n)) to resolve the issue. Now it works well on WSL. This is a common problem that Windows users may encounter.

Reply:
I think ebpf agent(AttachedEvent) is highly relying on Linux. I even doubt MacOS would work.
Don't try to depend on one Platforma, sometimes, it just doesn't work from one case to another.

Reply:
@mrproliu  Help!!!
I've been trying to do e2e testing for several days now, trying various methods such as using swctl to query with a single sentence, and also modifying service-instance.yml to adapt to the return value, but when it comes to the process list step, it always returns [].

I have not been able to successfully execute the test, and the whole process has been stuck in this part, making progress extremely difficult.

I hope you can help me figure out how to make this part run smoothly, and what kind of environment or tools I need to use. I have tried almost all the methods that I can try now.

My usage is Docker Desktop, and the instance is running normally from Containers.

Below is the original error message from the main branch:
----------
```
  ERROR failed to verify the output: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql instance list --service-name=service, error:
mismatch (-want +got):
  []interface{}{
-       map[interface{}]interface{}{
-               string("attributes"):   []interface{}{},
-               string("id"):           string("c2VydmljZQ==.1_dGVzdA=="),
-               string("instanceuuid"): string("c2VydmljZQ==.1_NTFjOGZmOTBjYmE2MTFlZGEwYjIwMjQyYWMxOTAwMDRAMTcyLjI1LjAuNA=="),
-               string("language"):     string("UNKNOWN"),
-               string("name"):         string("test"),
-       },
+       map[interface{}]interface{}{
+               string("attributes"): []interface{}{
+                       map[interface{}]interface{}{string("name"): string("Process No."), string("value"): string("1")},
+                       map[interface{}]interface{}{string("name"): string("hostname"), string("value"): string("e7786ddc74c7")},
+                       map[interface{}]interface{}{string("name"): string("OS Name"), string("value"): string("linux")},
+                       map[interface{}]interface{}{string("name"): string("ipv4s"), string("value"): string("172.25.0.4")},
+               },
+               string("id"):           string("c2VydmljZQ==.1_NTFjOGZmOTBjYmE2MTFlZGEwYjIwMjQyYWMxOTAwMDRAMTcyLjI1LjAuNA=="),
+               string("instanceuuid"): string("c2VydmljZQ==.1_NTFjOGZmOTBjYmE2MTFlZGEwYjIwMjQyYWMxOTAwMDRAMTcyLjI1LjAuNA=="),
+               string("language"):     string("GO"),
+               string("name"):         string("51c8ff90cba611eda0b20242ac190004@172.25.0.4"),
+       },
  }
```